src/sqlfluff/dialects/dialect_hive.py

```diff
@@ -556,3 +559,30 @@ class IntervalExpressionSegment(BaseSegment):
             ),
         ),
     )
+
+
+@hive_dialect.segment()
+class MsckRepairTableStatementSegment(BaseSegment):
+    """A repair table statement.
+
+    `MSCK REPAIR` reference here within Language Manual DDL page:
+    https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL
+    """
+
+    type = "msck_repair_table_statement"
+
+    match_grammar = Sequence(
+        "MSCK",
+        "REPAIR",
```

Comment:
```suggestion
        Ref.keyword("REPAIR", optional=True),
```

Reply:
REPAIR is optional in Hive
```
MSCK [REPAIR] TABLE table_name [ADD/DROP/SYNC PARTITIONS];
```

Reply:
Without repair, this is a different operation and wouldn't fit under the `RepairTableStatementSegment`. Check out comment above but another approach could be to have `MsckTableStatementSegment` for Hive that includes both then just have a 'MsckRepairTableStatement` for Spark.

Reply:
I'm personally leaning towards `MsckTableStatementSegment` that captures both in Hive and `MsckRepairTableStatementSegment` in Spark

Reply:
Added MsckTableStatementSegment in Hive dialect to address feedback